src/components/coinStats/CoinStats.js

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCoin, setSingleCoin } from '../../redux/coins/coinSlice';
+import '../coin.css';
+import Coinstat from './coinstat';
+
+function CoinStats() {
+  const { coinId } = useParams();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCoin(coinId));
+  }, [dispatch, coinId]);
+
+  useEffect(() => {
+    // Dispatch an action to update the singleCoin value
+    dispatch(setSingleCoin(/* new singleCoin value */));
+  }, [dispatch]);
+
+  const { singleCoin } = useSelector((state) => state.coins);
```

Comment:
[OPTIONAL] I noticed here that you dispatch an action to fetch information from the API for a single coin, then you set that single coin in the store. You can avoid sending extra requests to the API (In general, the less requests you have to send the better) in this case because you already have the list of coins in the store. So instead of sending a request to fetch a single coin, you can just filter the list of coins after getting it with `useSelector` and get the needed item and pass it to the `Coinstat` component.